### ISSUE TITLE
Client Credentials Quickstart tutorial appears to have a new bug

### ISSUE THREAD

**Issue by davidbarrows** [IdentityServer/IdentityServer4#4367]
**We can only help you if you are on the latest version. Either use the latest 2.x or 3.x**

Please only use the issue tracker for bug reports and/or feature requests. For general security questions, or free or commercial support options do __not__ use the issue tracker and instead see [here](http://docs.identityserver.io/en/latest/intro/support.html) for more details.

For bug reports,  include the relevant log files related to your issue. See here how to enable [logging](https://identityserver4.readthedocs.io/en/release/topics/logging.html). Delete this line once you have.

Finally, please keep the issue concise and to the point. If you paste in more code than the text for the issue you are reporting then we will most likely not read it. 

### Issue / Steps to reproduce the problem

- Go through the first Quickstart as described in the IdentityServer4 documentation.  When you get to the point where it says 

"That’s it your identity server should now be configured. If you run the server and navigate the browser to http://localhost:5000/.well-known/openid-configuration, you should see the so-called discovery document."

- it fails.  At first glance I suspect perhaps the reason it fails is because the launchSettings.json file that's created by the template looks like this:

```
{
  "profiles": {
    "SelfHost": {
      "commandName": "Project",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "applicationUrl": "https://localhost:5001"
    }
  }
}
```
Whereas the one in the github samples looks like this:

```
{
  "iisSettings": {
    "windowsAuthentication": false,
    "anonymousAuthentication": true,
    "iisExpress": {
      "applicationUrl": "http://localhost:5000",
      "sslPort": 0
    }
  },
  "profiles": {
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "SelfHost": {
      "commandName": "Project",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "applicationUrl": "http://localhost:5000"
    }
  }
}
```
This seems to be a recent breakage because I've done this tutorial in the past and it worked.

Thank you, -Dave (recent attendee at NDC London)

### Relevant parts of the log file

```
<log goes here>
```

**Comment by leastprivilege** [comment]
True - we switched everything over to https and 5001. Want to send PRs along the way?

**Comment by davidbarrows** [comment]
hi Dominic, sure, I can give it a try - would you like me to try to fix the documentation, or the template (or both)?  cheers

**Comment by leastprivilege** [comment]
I think it makes the most sense to make sure that the quick starts are aligned with the templates.

**Comment by davidbarrows** [comment]
It looks like my first PR will be to just update the documentation of the client credentials quickstart - to show https / 5001; should I make a PR off the master branch or perhaps the "releases/3.1.x" branch?  thx

**Comment by davidbarrows** [comment]
(actually it looks like the samples probably need updating also in the interest of making it consistent with the templates; e.g this file -> https://github.com/IdentityServer/IdentityServer4/blob/master/samples/Quickstarts/1_ClientCredentials/src/IdentityServer/Properties/launchSettings.json  )

**Comment by leastprivilege** [comment]
They belong to the docs, yes.

**Comment by davidbarrows** [comment]
Sorry for all the questions - maybe email would be easier - anyway in the old client credentials quickstart, it seemed you were using port 5000 for IdentityServer and port 5001 for the API.  Shouldn't they remain on separate ports?  (it seems in the new version of the template as we've been discussing, you've switched to https://localhost:5001 for IdentityServer; but I wonder if that should have stayed as 5000.  Because there's also the reference to the authority in the Startup.cs of the API ->  https://github.com/IdentityServer/IdentityServer4/blob/master/samples/Quickstarts/1_ClientCredentials/src/Api/Startup.cs

Instinctively I feel perhaps IdentityServer should be on https://localhost:5000 for the 1_ClientCredentials samples and templates, whereas the API should be on http://localhost:5001.  But maybe it's easier (since you've already changed the template) to just have the quickstart instructions and samples tell the developer to put the API on a different port (e.g. 5002).  

I have now done some further testing and made the following observations, which I suggest should be reflected in the docs:
- at the point where you tell the user to add the API project, once they've added it, then in Visual Studio they should set the solution to "multiple startup projects".
- Once multiple startup projects  I observed it does not work to set the API and IdentityServer projects to the same ports in launchSettings.  The solution fails to run.
- It does not run if both the API and IdentityServer are set to 5001.
- It errors if I set iisExpress to https://localhost:5001 - it says "unable to connect to web server IIS Express"
- However if I set the API to http://localhost:5001 and the "Api" profile to "http://localhost:5001" (whilst IdentityServer is set to https://localhost:5000), and with multiple startup projects, everything works.  I get the 401 as described in the quickstart (at the point just after having added the API project).

In summary my suggestion would be to either set the template back to port 5000, or tell the user to set the API port to 5002 or some other port.  (The latter is probably easier since I can just change the docs and samples).  Please let me know your thoughts, thanks.  -dB

**Comment by davidbarrows** [comment]
Hi Dominic - okay I tried to push a PR but permission denied.  My changes would be as follows:
- update the screenshot to show the discovery document is on https://localhost:5001
- update the quickstart documentation (RST file) to be consistent with template now running on 5001
- let the API run on http://localhost:5000 to differentiate it from the IdentityServer running on 5001; update the samples and documentation to be consistent with that change
- some additional minor cleanup to the docs

Please grant permission if you'd like me to push this PR.  Many thanks, -Dave Barrows

**Comment by leastprivilege** [comment]
You don't need any permission for that...

**Comment by leastprivilege** [comment]
..I would put the API on port 6000 using HTTPS

**Comment by davidbarrows** [comment]
thanks for your reply.  Two things:

1. I'm trying to create a new branch and push to it, so I can then do a PR; I'm getting the following "permission denied" error (do I need to fork instead of trying to create a branch?):

- "C:\Program Files\Git\bin\git.exe" push -u --recurse-submodules=check --progress "origin" refs/heads/bugfix/quickstart-documentation-fixes:refs/heads/bugfix/quickstart-documentation-fixes remote: **Permission to IdentityServer/IdentityServer4.git denied to davidbarrows.
fatal: unable to access 'https://github.com/IdentityServer/IdentityServer4.git/': The requested URL returned error: 403**
Done

Press Enter or Esc to close console...

2. on a Windows 10 machine, I'm not able to set the API port to https://localhost:6000 and get it to run.  Various problems occur including:

- Chrome considers 6000 to be an unsafe port.
- VS 2019 errors out saying "unable to connect to web server IIS Express".
- It doesn't actually work - even if I set IIS Express profile in launchSettings.json to http, but then set the API profile to https, it doesn't hit (Chrome says "the site can't be reached" and when I try to go to that URL - https://localhost:6000/identity - it acts like the endpoint doesn't exist.
- It runs fine when I set it to http://localhost:6000, but not https.  At that point it behaves consistently with what the Quickstart documentation says, and returns a 401 forbidden as expected.
- So please let me know what to do about the permissions issue with the git repo
- Then I can push my code and documentation changes to my branch and perhaps you can take a look.  I'd also be curious to know if this is just a problem for me and my setup... but I think running VS 2019 on a Windows 10 machine and testing with Google Chrome is a common setup for .Net devs.

**Comment by leastprivilege** [comment]
you fork - create a branch and then do the PR.

**Comment by leastprivilege** [comment]
I removed IIS express from all my launchsettings. If you need to run on self host and IIS with HTTPS I think you need to use a port like 44300 and up.

**Comment by davidbarrows** [comment]
please see my PR... I suggest port 6001 because when I tried using 6000, I get an "unsafe port" error using Chrome for Windows.  Using 6001 works around that problem.  Let me know if you'd like me to make any changes.  thanks, -Dave
